web/docs/understand/data-model/type-system.md

```diff
@@ -6,35 +6,298 @@ sidebar_position: 0
 
 One [design goal](/docs/understand/architecture/design-goals) of VAST is
 being expressive enough to capture the semantics of the domain. This led us to
-develop a rich type system for structured security data, inspired by the data
-model of the [Zeek](https://zeek.org) network security monitor.
+develop a rich type system for structured security data, inspired by the
+[Zeek](https://zeek.org) network security monitor.
+
+## Terminology
+
+The diagram below illustrates the type system at a glance:
+
+![Type System - VAST](type-system-vast.excalidraw.svg)
+
+All type have an optional name. They coexist in a global namespace.
+
+### Basic Types
+
+Basic types are stateless types with a static structure. Their representation
+is known a-priori.
+
+The following tables summarizes the basic types.
+
+| Type       | Description
+| ---------- | --------------------------------------
+| `none`     | Denotes an absent or invalid value
+| `bool`     | A boolean value
+| `int64`    | A 64-bit signed integer
+| `uint64`   | A 64-bit unsigned integer
+| `double`   | A 64-bit double (IEEE 754)
+| `duration` | A time span (nanosecond granularity)
+| `time`     | A time point (nanosecond granularity)
+| `string`   | A sequence of characters
+| `pattern`  | A regular expression
+| `ip`       | An IPv4 or IPv6 address
+| `subnet`   | An IPv4 or IPv6 subnet
+
+### Complex Types
+
+Complex types are stateful types that carry additional runtime information.
+
+#### Enumeration
+
+The `enum` type is a list of predefined string values. It comes in handy for
+low-cardinality values from a fixed set of options.
+
+VAST implements an `enum` as an Arrow Dictionary.
+
+#### List
+
+The `list` type is an ordered sequence of values with a fixed element type.
+
+Lists have zero or more elements.
+
+#### Record
+
+The `record` type consists of an ordered sequence *fields*, each of which have a
+name and type. Records must have at least one field.
+
+The field name is an arbitrary UTF-8 string.
+
+The field type is any VAST type.
+
+Optionality of a field is reflected by its type.
```

Comment:
```suggestion
```

Optional fields have the value null, so I would just omit this line. It doesn't add any value with the optionality section right below.